source/openqasm/openqasm3/printer.py

```diff
@@ -0,0 +1,700 @@
+import contextlib
+import dataclasses
+import io
+
+from typing import Sequence, Optional
+
+from . import ast, properties
+from .visitor import QASMVisitor
+
+__all__ = ("dump", "dumps")
+
+
+def dump(node: ast.QASMNode, file: io.TextIOBase, **kwargs):
```

Comment:
A Docstring would be nice since this is an exported function.

Reply:
Whoops, I forgot to write this.  Thanks for pointing it out - I've done the documentation completely in ca19088.